mlrun/api/api/endpoints/workflows.py

```diff
@@ -0,0 +1,287 @@
+                    project=project.metadata.name,
+                    name=name,
+                    json=workflow_request.dict(),
+                )
+            workflow_action = "schedule"
+            mlrun.api.crud.WorkflowRunners().schedule(
+                runner=workflow_runner,
+                project=project,
+                workflow_request=updated_request,
+                db_session=db_session,
+                auth_info=auth_info,
+            )
+            status = "scheduled"
+
+        else:
+            workflow_action = "run"
+            run = mlrun.api.crud.WorkflowRunners().run(
+                runner=workflow_runner,
+                project=project,
+                workflow_request=updated_request,
+            )
+            status = mlrun.run.RunStatuses.running
+            run_uid = run.uid()
+    except Exception as error:
+        logger.error(traceback.format_exc())
+        log_and_raise(
+            reason=f"Workflow {workflow_spec.name} {workflow_action} failed!, error: {error}"
+        )
+
+    return mlrun.api.schemas.WorkflowResponse(
+        project=project.metadata.name,
+        name=workflow_spec.name,
+        status=status,
+        run_id=run_uid,
+        schedule=workflow_spec.schedule,
+    )
+
+
+@router.get(
+    "/projects/{project}/{name}/{uid}",
+    response_model=mlrun.api.schemas.GetWorkflowResponse,
+)
+def get_workflow_id(
+    project: str,
+    name: str,
+    uid: str,
+    auth_info: mlrun.api.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: Session = fastapi.Depends(mlrun.api.api.deps.get_db_session),
+) -> mlrun.api.schemas.GetWorkflowResponse:
+    """
+    Retrieve workflow id from the uid of the workflow runner.
+    When creating a remote workflow we are creating an auxiliary function
+    which is responsible for actually running the workflow,
+    as we don't know beforehand the workflow uid but only the run uid of the auxiliary function we ran,
+    we have to wait until the running function will log the workflow id it created.
+    Because we don't know how long it will take for the run to create the workflow
+    we decided to implement that in an asynchronous mechanism which at first,
+    client will get the run uid and then will pull the workflow id from the run id
+    kinda as you would use a background task to query if it finished.
+    Supporting workflows that executed by the remote engine **only**.
+
+    :param project:     name of the project
+    :param name:        name of the workflow
+    :param uid:         the id of the running job that runs the workflow
+    :param auth_info:   auth info of the request
+    :param db_session:  session that manages the current dialog with the database
+
+    :returns: workflow id
+    """
+    # Check permission READ run:
+    mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
```

Comment:
this has become async operation. thus, your function decorator must be `async` as well. see my comment above in case the rest of the function logic must run in a threadpool.

Reply:
fixed as well.